lib/src/command/barback.dart

```diff
@@ -24,6 +25,27 @@ abstract class BarbackCommand extends PubCommand {
   /// The build mode.
   BarbackMode get mode => new BarbackMode(argResults["mode"]);
 
+  // The current compiler mode.
+  Compiler get compiler {
+    if (argResults.options.contains('dart2js') &&
+        argResults.wasParsed('dart2js')) {
+      if (argResults["compiler"] && argResults.wasParsed("compiler")) {
+        throw new ArgumentError(
+            "The `dart2js` arg can't be used with the `compiler` arg. Prefer "
+            "using the compiler flag.");
+      }
+      if (argResults['dart2js']) {
+        return Compiler.dart2Js;
+      } else {
+        return Compiler.none;
+      }
+    } else if (argResults.options.contains("compiler")) {
+      return Compiler.byName(argResults["compiler"]);
```

Comment:
Added a test - it gives exit code 64 and prints a message about it being an invalid option (but no stack trace)